plugin/src/main/kotlin/com/nishtahir/CargoBuildTask.kt

```diff
@@ -162,3 +176,86 @@ open class CargoBuildTask : DefaultTask() {
         }.assertNormalExitValue()
     }
 }
+
+// We don't want to invoke `rustc -Vv` a ton, but we can't invoke it at all without
+// the gradle project (despite this not being dependent on the project). As a result,
+// we need to do some locking in order to cache the result of that.
+//
+// This object exists to encapsulate that, it just exposes a `fun getValue(p: Project): String?`.
+//
+// TODO: Is there somewhere more natural to put configuration state like this?
+private object DefaultTargetTriple {
+    // Initialized once, write-protected by `writeLock`, but no lock needs to be held when reading.
+    @Volatile
+    private var knownValue: String? = null
+
+    // Protects `knownValue` during writing. No protection is needed during reading. We also hold it
+    // while computing the value, to avoid wasting work.
+    private val writeLock = Object()
```

Comment:
I think Kotlin's lazy initialization provides the locking you want -- see https://kotlinlang.org/docs/reference/delegated-properties.html#lazy.  `lateinit` as well.

Reply:
I couldn't figure out how to use these given the requirement of the `project`. If we don't need the project, I believe it can just be a static ctor which will do the locking also.